packages/jellyfish-api-core/src/category/masternode.ts

```diff
@@ -248,6 +265,26 @@ export interface CreateMasternodeOptions {
   timelock?: MasternodeTimeLock
 }
 
+type UpdateMasternodeOptions = UpdateMasternodeOptions1 | UpdateMasternodeOptions2 | UpdateMasternodeOptions3
```

Comment:
I use this type because one of the three parameters must be passed in.

I could use the following code.

```
type UpdateMasternodeOptions = AtLeastOne<Options>;

type AtLeastOne<T> = {
  [K in keyof T]: Pick<T, K> & Partial<Omit<T, K>>;
}[keyof T];

export interface Options {
  ownerAddress: string
  operatorAddress: string
  rewardAddress: string
}
```

But I feel like this code is a bit complex and I stick with old code to increase code readibility

Reply:
I think perhaps in this case, the complexity of `AtLeastOne` is valid as it simplifies the interface declaration scope. The naming of the type is also informative as to its intent, so if it works as expected I would 👍🏻 your custom wrapper type.

It's a pretty neat solution, TypeScript is just going to introduce some complexities from time to time. 😸

Reply:
I spend a few days to think and I agree with your suggestion.